plugins/Observability/src/Observability.cpp

```diff
@@ -1838,6 +1613,273 @@ void Observability::showReport(bool b)
+					int selday2 = 0;
+					bool bestBegun = false; // Are we inside a good time range?
+					bool atLeastOne = false;
+					QString dateRange;
+					bool poleNight, twiGood;
+
+					for (int i=0; i<nDays; i++)
+					{
+						poleNight = sunSidT[0][i]<0.0 && qAbs(sunDec[i]-mylat)>=halfpi; // Is it night during 24h?
+						twiGood = (poleNight && qAbs(objectDec[i]-mylat)<halfpi)?true:CheckRise(i);
+
+						if (twiGood && bestBegun == false)
+						{
+							selday = i;
+							bestBegun = true;
+							atLeastOne = true;
+						}
+
+						if (!twiGood && bestBegun == true)
+						{
+							selday2 = i;
+							bestBegun = false;
+							if (selday2 > selday)
+							{
+								// FIXME: This kind of concatenation is bad for i18n.
+								if (!dateRange.isEmpty())
+									dateRange += ", ";
+								dateRange += QString("%1").arg(formatAsDateRange(selday, selday2));
+							}
+						}
+					}
+
+					// Check if there were good dates till the end of the year.
+					if (bestBegun)
+					{
+						// FIXME: This kind of concatenation is bad for i18n.
+						 if (!dateRange.isEmpty())
+							 dateRange += ", ";
+						dateRange += formatAsDateRange(selday, 0);
+					}
+
+					if (dateRange.isEmpty())
+					{
+						if (atLeastOne)
+						{
+							//ObsRange = q_("Observable during the whole year.");
+							lineObservableRange = msgWholeYear;
+						}
+						else
+						{
+							//ObsRange = q_("Not observable at dark night.");
+							lineObservableRange = msgNotObs;
+						}
+					}
+					else
+					{
+						// Nights when the target is above the horizon
+						lineObservableRange = msgAboveHoriz.arg(dateRange);
+					}
+				} // Comes from show_Good_Nights==True"
+			} // Comes from the "else" of "culmAlt>=..."
+}
+
+void Observability::renderResults() {
+   StelCore *core = StelApp::getInstance().getCore();
+   // Set the painter:
+   StelPainter painter(core->getProjection2d());
+   painter.setColor(fontColor[0], fontColor[1], fontColor[2], 1.f);
+   font.setPixelSize(fontSize);
+   painter.setFont(font);
+
+// Print all results:
+	StelProjector::StelProjectorParams params = core->getCurrentStelProjectorParams();
+	float ppx = static_cast<float>(params.devicePixelsPerPixel);
+	int lineSpacing = static_cast<int>(ppx * 1.3f * fontSize);  // between lines
+//	int groupSpacing = static_cast<int>(6*fontSize*ppx);  // between daily and yearly results
+	int yLine = static_cast<int>(8*fontSize*ppx + 50*ppx);
+	int xLine = 80*ppx;
+
+//	if (show_Today)
+//	{
+//		//renderer->drawText(TextParams(xLine, yLine,q_("TODAY:")));
+//		painter.drawText(xLine, yLine, msgToday);
+//		painter.drawText(xLine + fontSize, yLine - lineSpacing, RS2);
+//		painter.drawText(xLine + fontSize, yLine - lineSpacing*2, RS1);
+//		painter.drawText(xLine + fontSize, yLine - lineSpacing*3, Cul);
+//		yLine -= groupSpacing;
+//	}
+
+	if ((isMoon(objectSelection) && show_FullMoon) || (!isSun(objectSelection) && !isMoon(objectSelection) && shouldShowYear()))
```

Comment:
objectSelection seems to be unknown. Did you push the latest version?

Reply:
Oh, whoops. Forgot to push that. Pushed.